src/josegonzalez/Queuesadilla/Engine/RabbitmqEngine.php

```diff
@@ -0,0 +1,325 @@
+<?php
+
+namespace josegonzalez\Queuesadilla\Engine;
+
+use Bunny\Client;
+use Exception;
+use josegonzalez\Queuesadilla\Engine\Base;
+
+class RabbitmqEngine extends Base
+{
+    protected $baseConfig = [
+        'database' => null,
+        'user' => 'guest',
+        'pass' => 'guest',
+        'persistent' => true,
+        'port' => 5672,
+        'queue' => 'default',
+        'host' => '127.0.0.1',
+        'timeout' => 0,
+        'insist' => false,
+        'login_method' => 'AMQPLAIN',
+        'login_response' => null,
+        'locale' => 'en_US',
+        'connection_timeout' => 3,
+        'read_write_timeout' => 3,
+        'context' => null,
+        'keepalive' => false,
+        'heartbeat' => 0,
+
+        'passive_queue' => false,
+        'durable_queue' => true,
+        'exclusive' => false,
+        'auto_delete_queue' => false,
+        'nowait' => false,
+        'arguments' => [],
+
+        'exchange' => 'jobs',
+        'type' => 'direct',
+        'passive_exchange' => false,
+        'durable_exchange' => true,
+        'auto_delete_exchange' => false,
+
+        'acknowledge' => true,
+
+        'confirm' => false,
+        'connect_attempt_delay' => 0.1,
+        'connect_attempts' => 3,
+        'mandatory_publish' => false,
+        'immediate_publish' => false,
+
+    ];
+
+    public $channel;
+
+    protected $handlerAttached = false;
+
+    protected $queues = null;
+
+    /**
+     * {@inheritDoc}
+     */
+    public function connect()
+    {
+        $vhost = '/';
+        $database = $this->config('database');
+        if ($database !== null) {
+            $vhost = '/' . $database;
```

Comment:
no need to wrap in an if, can just always append the slash with the config value